### ISSUE TITLE
“keras.backend.variable” is not behaving correctly in keras as opposed to tensorflow

### ISSUE THREAD

**Issue by ehsanhaghighat** [keras-team/keras#13106]
I want to define **trainable scalar parameters** in my models. In TensorFlow, this is done using `tf.Variable`. In Keras, `keras.backend.variable` is supposed to behave the same way. However, when I use `model.fit`, `keras` does not change the `variable` during the optimization process. Does anyone know why? 

To test, please uncomment `RUN_ON = "tensorflow"` or `RUN_ON = "keras"` to run on either of engines. 

    import numpy as np
    import keras as k
    import tensorflow as tf
    import matplotlib.pyplot as plt

    # RUN_ON = "tensorflow"
    # RUN_ON = "keras"

    
    b_true = 3.0
    w_true = 5.0
    x_true = np.linspace(0.0, 1.0, 1000).reshape(-1, 1)
    y_true = x_true * w_true + b_true
    ids = np.arange(0, x_true.shape[0])
    
    if RUN_ON=="keras":
        x = k.Input((1,), dtype="float32", name="x")
        Fx = k.layers.Dense(1, use_bias=False, name="Fx")(x)
        b = k.backend.variable(1.0, name="b")
        y = k.layers.Lambda(lambda x: x+b, name="Add")(Fx)
    
        model = k.Model(inputs=[x], outputs=[y])
        model.compile("adam", loss="mse")
        # model.summary()
    
        model.fit(x_true, [y_true], epochs=100000, batch_size=1000)
        y_pred = model.predict(x_true)
    
    elif RUN_ON=="tensorflow":
        x = tf.placeholder("float32", shape=[None, 1], name="x")
        Fx = tf.layers.Dense(1, use_bias=False, name="Fx")(x)
        b = tf.Variable(1.0, name="b")
        y = Fx + b
        yp = tf.placeholder("float32", shape=[None, 1], name="y")
    
        loss = tf.reduce_mean(tf.square(yp - y))
        opt = tf.train.AdamOptimizer(0.001).minimize(loss)
    
        with tf.Session() as sess:
            sess.run(tf.global_variables_initializer())
            for i in range(100000):
                np.random.shuffle(ids)
                opt_out, loss_val, b_val = sess.run([opt, loss, b], feed_dict={x: x_true[ids], yp: y_true[ids]})
                print("epoch={:d}  loss={:e}   b_val={:f}".format(i, loss_val, b_val))
                if loss_val < 1.0e-9:
                    break
    
            y_pred = sess.run([y], feed_dict={x: x_true, yp: y_true})[0]
    
    else:
        raise ValueError('`RUN_ON` should be either `keras` or `tensorflow`.')
    
    plt.plot(x_true, y_true, '--b', linewidth=4)
    plt.plot(x_true, y_pred, 'r')
    plt.show()
    #
